File: app/src/customer/java/com/example/shoppingapp_customer/adapters/SearchItemAdapter.java
package com.example.shoppingapp_customer.adapters;

import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.core.util.Pair;
import androidx.recyclerview.widget.RecyclerView;

import com.example.shoppingapp.Item;
import com.example.shoppingapp_customer.MainActivity2;

import com.example.shoppingapp.R;
import com.example.shoppingapp_customer.fragments.DescriptionScreen;

import java.util.List;
import java.util.Random;

import timber.log.Timber;

public class SearchItemAdapter extends RecyclerView.Adapter<SearchItemAdapter.SearchItemViewHolder> {

    Context context;
    List<Item> itemList;
    Random rnd = new Random();
    int maxValue=180;

    public SearchItemAdapter(Context context) {
        this.context = context;
    }

    public SearchItemAdapter(Context context, List<Item> itemList) {
        this.context = context;
        this.itemList = itemList;
    }

    @NonNull
    @Override
    public SearchItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater=LayoutInflater.from(context);
        View v=layoutInflater.inflate(R.layout.item_search_layout,parent,false);
        return new SearchItemAdapter.SearchItemViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull SearchItemViewHolder holder, int position)
    {
        int color=Color.argb(130, rnd.nextInt(maxValue), rnd.nextInt(maxValue), rnd.nextInt(maxValue));
        Timber.e(String.valueOf(color));
        holder.itemLayout.getBackground().setTint(color);


        holder.itemLayout.setOnClickListener(v-> {
            Intent intent=new Intent(context, MainActivity2.class);
            intent.putExtra(context.getString(R.string.FRAGMENT_NAME_TAG), DescriptionScreen.class.getSimpleName());
            intent.putExtra(context.getString(R.string.COLOR_TAG),color);
//            intent.putExtra(context.getString(R.string.OBJECT_TAG), itemList.get(position));

            context.startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return 10;
    }

    class SearchItemViewHolder extends RecyclerView.ViewHolder{

        ConstraintLayout itemLayout;
        ImageView itemImg;
        ImageView cartImg;
        ImageView bookmarkImg;
        TextView itemNameTxt;
        TextView itemPriceTxt;
        TextView itemQuantityTxt;

        public SearchItemViewHolder(@NonNull View itemView) {
            super(itemView);

            itemLayout = itemView.findViewById(R.id.search_item_layout);
            itemImg = itemView.findViewById(R.id.search_item_img);
            cartImg = itemView.findViewById(R.id.search_item_cart_img);
            bookmarkImg = itemView.findViewById(R.id.search_item_bookmark_img);
            itemNameTxt = itemView.findViewById(R.id.search_item_name_txt);
            itemPriceTxt = itemView.findViewById(R.id.search_item_price_txt);
            itemQuantityTxt = itemView.findViewById(R.id.search_item_quantity_txt);
        }
    }
}

File: app/src/customer/java/com/example/shoppingapp_customer/adapters/SearchItemAdapter.kt
package com.example.shoppingapp_customer.adapters

import android.content.Context
import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.navigation.findNavController
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.shoppingapp.*
import com.example.shoppingapp.util.Firebase
import com.example.shoppingapp_customer.adapters.SearchItemAdapter.SearchItemViewHolder
import timber.log.Timber
import timber.log.Timber.tag
import java.util.*

class SearchItemAdapter(val context: Context,var itemList:MutableList<Item>) : RecyclerView.Adapter<SearchItemViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SearchItemViewHolder {
        val layoutInflater = LayoutInflater.from(context)
        val v = layoutInflater.inflate(R.layout.item_search_layout, parent, false)
        return SearchItemViewHolder(v)
    }

    override fun onBindViewHolder(holder: SearchItemViewHolder, position: Int) {
        holder.txtPrice.text=itemList[position].price+" $"
        holder.txtName.text=itemList[position].name
        holder.layout.setOnClickListener {
            val bundle= Bundle()
            bundle.putString(context.getString(R.string.PASS_ITEM_ID_KEY),itemList[position].id)
            it.findNavController().navigate(R.id.action_itemsScreen_to_descriptionScreen,bundle)
        }
        Glide.with(context)
                .load(itemList[position].icon)
                .error(R.drawable.error)
                .into(holder.img)

        view(holder,(Cart.getIndex(itemList[position].id)>=0))

        holder.layoutAddToCart.setOnClickListener {
            when {
                (Cart.getIndex(itemList[position].id)>=0) -> {
                    Firebase.removeItemFromCart(context,CartItem(itemList[position]))
                    view(holder,false)
                    tag("SearchAdapter").d("removeFromCart")
                    notifyDataSetChanged()
                }
                itemList[position].stock.toInt()==0 -> {
                    Toast.makeText(context,"No items in stock\ntry again later", Toast.LENGTH_LONG).show()
                    tag("SearchAdapter").d("empty")
                }
                else -> {
                    tag("SearchAdapter").d("addToCart")
                    Firebase.addToCart(context, CartItem(itemList[position]))
                    view(holder,true)
                    notifyDataSetChanged()
                }
            }
        }

    }

    fun view(holder: SearchItemViewHolder,inCart:Boolean){
        if(inCart){
            holder.cartImg.setImageResource(R.drawable.img_check)
            holder.txtAddToCart.text = context.getString(R.string.in_cart)
            holder.layoutAddToCart.setBackgroundColor(context.getColor(R.color.in_cart_color))
        }else{
            holder.cartImg.setImageResource(R.drawable.img_add_to_cart)
            holder.txtAddToCart.text = context.getString(R.string.add_to_cart)
            holder.layoutAddToCart.setBackgroundColor(context.getColor(R.color.btn_color))
        }
    }

    override fun getItemCount(): Int {
        return itemList.size
    }

    class SearchItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var txtName:TextView
        var txtPrice:TextView
        var img:ImageView
        var layout:ConstraintLayout
        init {
            this.txtName = itemView.findViewById(R.id.txt_item_name_isl)
            this.txtPrice = itemView.findViewById(R.id.txt_item_price_isl)
            this.layout = itemView.findViewById(R.id.layout_item_isl)
            this.img = itemView.findViewById(R.id.img_isl)
        }

        var cartImg:ImageView
        var layoutAddToCart:ConstraintLayout
        var txtAddToCart:TextView

        init{
            this.txtAddToCart = itemView.findViewById(R.id.txt_add_to_cart_isl)
            this.cartImg = itemView.findViewById(R.id.img_cart_isl)
            this.layoutAddToCart = itemView.findViewById(R.id.layout_add_to_cart_isl)
        }
    }
}